onnx/shape_inference/implementation.cc

```diff
@@ -78,6 +70,13 @@ std::string GetElemTypeString(const TypeProto_SparseTensor& type) {
 inline bool IsOnnxDomainOp(const NodeProto& node, const std::string& op_type) {
   return (IsOnnxDomain(node.domain()) && (node.op_type() == op_type));
 }
+
+void UnknownOpError(const NodeProto& nodeProto) {
+  if (checker::check_is_experimental_op(nodeProto)) {
+    fail_type_inference("Experimental operator '", nodeProto.op_type(), "' no longer upported.");
+  }
+  fail_type_inference("Unknown operator/function: '", nodeProto.op_type(), "' in domain: '", nodeProto.domain(), "'.");
```

Comment:
Will this fail custom ops? What do we expect when users run this on a model that contains them? Should we allow an option to skip them?

Reply:
@titaiwangms @BowenBao

Reply:
Can users register custom op schema through ONNX api? If a customer has a set of common custom ops, they might want checker to guard them as well.

Reply:
> Or, at least, indicate the list of custom-ops (instead of being silent for every undefined op).

Vote to have warning message.

Reply:
> > Or, at least, indicate the list of custom-ops (instead of being silent for every undefined op).
> 
> Vote to have warning message.

I don't understand: do you mean a warning message, instead of an error? (We don't have any such distinction yet.) 

I am undoing this specific change, and will let the special-handling continue as before. However, that behavior seems somewhat fragile and worth fixing. But better to keep that change separate from the other changes in this PR.